src/Summoner/Project.hs

```diff
@@ -51,12 +49,12 @@ generateProject projectName Config{..} = do
     email       <- queryDef "Maintainer e-mail: " cEmail
     putText categoryText
     category <- query "Category: "
-    license  <- choose "License: " $ map unLicense $ ordNub (cLicense : licenseNames)
+    license  <- choose "License: " $ map showLicense universe
 
     -- License creation
-    let licenseGithub = snd
-                      $ Unsafe.head
-                      $ dropWhile ((/= license) . unLicense . fst) githubLicenseQueryNames
+    let licenseGithub = case parseLicense license of 
+            Just l -> githubLicenseQueryNames l
+            Nothing -> error "Unrecognised license name"
```

Comment:
This case shouldn't ever happen, but yeah, we can't prove that to compiler..

I'm now thinking about patching the `choose` function so it will take the list of `[a]` instead of `[Text]` and return the `a` (instead of `Text`). This will make the code safer and remove this unnecessary check. What do you think?

Reply:
> This case shouldn't ever happen, but yeah, we can't prove that to compiler..

It happened when I was wrong :)

Reply:
I agree with @vrom911. I think it will be good if `choose` could take list of `a` and return `a` on success.